lib/pxenforcer.js

```diff
@@ -76,13 +76,21 @@ class PxEnforcer {
             return cb();
         }
 
-        if (userAgent && this._config.FILTER_BY_USERAGENT.includes(userAgent)) {
-            // user agent is filtered
-            this.logger.debug(`Skipping verification for filtered user agent ${userAgent}`);
-            return cb();
+        if (userAgent && this._config.FILTER_BY_USERAGENT && this._config.FILTER_BY_USERAGENT.length > 0) {
+            for (const ua of this._config.FILTER_BY_USERAGENT) {
+                if (ua instanceof RegExp && userAgent.match(ua)) {
+                    this.logger.debug(`Skipping verification for filtered user agent ${userAgent}`);
+                    return cb();
+                }
+
+                if (typeof ua === 'string' && userAgent.toLowerCase() === ua.toLowerCase()) {
+                    this.logger.debug(`Skipping verification for filtered user agent ${userAgent}`);
+                    return cb();
+                }
```

Comment:
Logic looks good, I'd just split this out into a separate function. I think in general the functions in this enforcer are very long so the more we can split them up the better.

Reply:
checking